core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -261,8 +261,8 @@ public State isWatching() {
     public boolean removeKey(ECKey key) {
         lock.lock();
         try {
-            boolean a = hashToKeys.remove(ByteString.copyFrom(key.getPubKeyHash())) != null;
-            boolean b = pubkeyToKeys.remove(ByteString.copyFrom(key.getPubKey())) != null;
+            boolean a = hashToKeys.remove(ByteBuffer.wrap(key.getPubKeyHash())) != null;
+            boolean b = pubkeyToKeys.remove(ByteBuffer.wrap(key.getPubKey())) != null;
```

Comment:
Needs array.copy here

Reply:
At first I was jumping at reverting this change for both lines, as we shouldn't track changes in a different class with decisions about code in this class but on the other hand ... we could protect ourselves from others corrupting the cached value by copying ... the potentially already corrupted cached value? That makes no sense. `getPubKey` requires the user to not alter it, as it would directly alter the cache. So we better promise to not corrupt it ourselves neither (we don't, do we?) and avoid drawing a copy here, too.

Reply:
Is there some `@PinkyPromiseToNotAlterObject` annotation somewhere, so my IDE can slap my fingers if I do?

Reply:
I suppose that annotations are great idea, but what if someone would use vim?